### PR TITLE
Dependency graph fix to compute dominators per function

### PR DESCRIPTION
**Pull request by danpoe**

### CODE REVIEW

File: src/goto-programs/goto_program_template.h [hunk 1]
```diff
@@ -265,6 +265,23 @@ class goto_program_templatet
     return t;
   }
 
+  inline static const irep_idt get_function_id(
```
Comment (tautschnig) [inline src/goto-programs/goto_program_template.h]: I don't think "inline" is necessary here (the definition is within the class definition).

File: src/analyses/dependence_graph.cpp [hunk 2]
```diff
@@ -260,22 +265,31 @@ void dep_graph_domaint::transform(
...
-        ++it;
+      control_dependencies(from, to, *dep_graph);
+    }
+    else
+    {
+      // edge to function entry point
+
+      dep_graph_domaint *s=
+        dynamic_cast<dep_graph_domaint*>(&(dep_graph->get_state(next)));
+      assert(s!=0);
```
Comment (peterschrammel) [inline src/analyses/dependence_graph.cpp]: use nullptr

Comment (tautschnig) [inline src/analyses/dependence_graph.cpp]: I've added #489 for a (future) general cleanup in this regard.

**Comment by danpoe** [comment]
Ok, I think this is ready to be merged.

**Comment by peterschrammel** [comment]
@danpoe, please rebase.

**Comment by danpoe** [comment]
The test slice16 was failing with these changes. It seems though that the problem is in the full slicer as the dependency graph output looks fine. I've marked the test as KNOWNBUG for now.

**Pull request merged**